### DISCUSSION TITLE
Failure in bootstrap-orbit.sh

### DISCUSSION THREAD

**Discussion by Vipin-Sharma** [google/orbit#4436] [Q&A]
Tried to build orbit on ubuntu desktop, facing the below error, complete log is available at [orbit_build_ubuntu_20_04_error.txt](https://github.com/google/orbit/files/6409638/orbit_build_ubuntu_20_04_error.txt)

ERROR: Failed requirement 'c-ares/1.15.0@conan/stable' from 'grpc/1.27.3@orbitdeps/stable'
ERROR: Unable to find 'c-ares/1.15.0@conan/stable#0' in remotes

Kernel version: 5.8.0-50-generic

is there any prerequisite I am missing here?

**Comment by crackcomm** [comment]
For me it rejects valid credentials with message "ERROR: Wrong user or password".

**Comment by beckerhe** [comment]
Hi both of you,
the build was broken for a couple of days due to the sunset of bintray.

Yesterday I had been migrating all the packages to the new Artifactory offering, so the build should be fixed.

Can you please update your main branch and try again? That should hopefully fix your problem. If not, please let me know here.

**Comment by shokrof** [comment]
I am receiving this error while building
`Using public remotes for conan.
ERROR: Error reading 'default_relwithdebinfo' profile: Profile not found: gcc7_relwithdebinfo
`

**Comment by beckerhe** [comment]
It seems your system's default compiler is pretty old (GCC7) and not supported by us.

Do you have another compiler or a more recent version of GCC available? GCC9 or 10 work well.
For clang version 7, 8 and 9 are supported.

You can explicitly target a certain compiler by calling `./build.sh <compiler_relwithdebinfo>`, for example `./build.sh clang9_relwithdebinfo` to compile with clang-9.

**Comment by Vipin-Sharma** [comment]
Tried to build it from the latest code and getting this error now.

```
abseil/20200923.3: Generator cmake created conanbuildinfo.cmake
abseil/20200923.3: Calling build()
/bin/sh: 1: cmake: not found
abseil/20200923.3: 
abseil/20200923.3: ERROR: Package '6eeea3904f53d9f7311f5a5b447089286767e0b4' build failed
abseil/20200923.3: WARN: Build folder /home/vipin/.conan/data/abseil/20200923.3/_/_/build/6eeea3904f53d9f7311f5a5b447089286767e0b4
ERROR: abseil/20200923.3: Error in build() method, line 60
	cmake = self._configure_cmake()
while calling '_configure_cmake', line 54
	self._cmake.configure()
	ConanException: Error 127 while executing cd '/home/vipin/.conan/data/abseil/20200923.3/_/_/build/6eeea3904f53d9f7311f5a5b447089286767e0b4' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DCONAN_CMAKE_CXX_STANDARD="17" -DCONAN_CMAKE_CXX_EXTENSIONS="OFF" -DCONAN_STD_CXX_FLAG="-std=c++17" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="gcc" -DCONAN_COMPILER_VERSION="9" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libstdc++11" -DCMAKE_INSTALL_PREFIX="/home/vipin/.conan/data/abseil/20200923.3/_/_/package/6eeea3904f53d9f7311f5a5b447089286767e0b4" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DABSL_ENABLE_INSTALL="True" -DBUILD_TESTING="False" -Wno-dev '/home/vipin/.conan/data/abseil/20200923.3/_/_/build/6eeea3904f53d9f7311f5a5b447089286767e0b4'

```

GCC version:

```
vipin@vipin-Z490M:~/githubprojects/orbit$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-17ubuntu1~20.04' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-HskZEa/gcc-9-9.3.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04) 
```

**Comment by beckerhe** [comment]
```
/bin/sh: 1: cmake: not found
```

Can you try installing `cmake`?

**Comment by shokrof** [comment]
@beckerhe thanks! I tried running ./build.sh gcc9_relwithdebinfo it passed the previous error but still failed to build the system because of not enough space(I have 4 GB) free space. I will try to free more space.
it is strange because I didn't have gcc7 installed on my computer and bootstrap.sh installed it and made it the default compiler. I am using ubuntu 18.04.

**Comment by beckerhe** [comment]
Hi Mostafa,
we don't explicitly install any compiler. If it was installed by bootstrap, then it was pulled in as an implicit dependency. You can try uninstalling it again using `apt`. If it's actually needed by any other package `apt` should complain about it.

Regarding space: You can try building a release build with`./build.sh gcc9_release` which does not retain any debug information. That might save you enough disk space to get through the build.

One more thing: In the past we had problems with the Qt version shipped by Ubuntu 18.04. We need at least Qt 5.12.4 and as far as I remember, 18.04 ships with 5.9. I believe you either have to update Qt or Ubuntu to succeed.

Cheers,
Henning

**Comment by Vipin-Sharma** [comment]
> ```
> 
> /bin/sh: 1: cmake: not found
> 
> ```
> 
> 
> 
> Can you try installing `cmake`?
> 
> 

It worked after installing cmake. Thanks!

**Comment by chadbrewbaker** [comment]
I'm getting a Conan python parse error at the end of the build :(. The build also barfed on default -j8 on my 16G RAM laptop under wsl2 when linking.  There needs to be a binary release for Ubuntu 20.04 on x86 and aarch64 that gets baked routinely.

**Comment by beckerhe** [comment]
Hi Chad,
can you share the exact error message?

- Henning

**Comment by hrishikeshtawade04** [comment]
> I am receiving this error while building
> `Using public remotes for conan. ERROR: Error reading 'default_relwithdebinfo' profile: Profile not found: gcc7_relwithdebinfo `

Encountered the same issue. I switched gcc to version 9 but it complains with the same error.

**Comment by beckerhe** [comment]
Can you try to call `./build.sh gcc9_release` to explicitly build with GCC9?

**Comment by GhislainViguier** [comment]
Hi, I'm also having troubles compiling Orbit on Ubuntu-20.04 with GCC9 and Orbit release/1.80
Here is the first error message :
```
./build.sh gcc9_release
[...]
In file included from /usr/include/c++/9/chrono:41,
                 from ./time.h:77,
                 from /usr/include/pthread.h:23,
                 from /usr/include/x86_64-linux-gnu/c++/9/bits/gthr-default.h:35,
                 from /usr/include/x86_64-linux-gnu/c++/9/bits/gthr.h:148,
                 from /usr/include/c++/9/ext/atomicity.h:35,
                 from /usr/include/c++/9/bits/ios_base.h:39,
                 from /usr/include/c++/9/ios:42,
                 from /usr/include/c++/9/ostream:38,
                 from /home/gys/.conan/data/abseil/20211102.0/_/_/build/89a41e343f2a5875e04cbefbed6c0730f1e8d38c/source_subfolder/absl/time/internal/cctz/include/cctz/civil_time_detail.h:20,
                 from /home/gys/.conan/data/abseil/20211102.0/_/_/build/89a41e343f2a5875e04cbefbed6c0730f1e8d38c/source_subfolder/absl/time/internal/cctz/src/civil_time_detail.cc:15:
/usr/include/c++/9/ctime:60:11: error: ‘::clock_t’ has not been declared
   60 |   using ::clock_t;
[...]
```
I have the same type of error with Clang-11.
Do I need to upgrade my Ubuntu to the last version?
